### PR TITLE
Add date to Todo

### LINKED ISSUE #138: Implement Todo - Add date to Todo and related commands

### LINKED ISSUE #139: Add tags to todo card

### LINKED ISSUE #140: Add icons to date and priority

### PR DESCRIPTION
**Pull request by Ugholaf**
# Changes
- Added `Date` to `Todo`.
- Include icons for the date and priority display field.
- Todo panel display will now display tags
- When starting the app, it will show today's todo.

# Todo Show Commands
- `todo show`: show all todos.
- `todo show today`: show todos with the date as the current date.
- `todo show date/DATE`: show todos with the date specified. 
- `todo show date/DATE1 to DATE2`: show todos from DATE1 to DATE2.
- `todo show t/TAG`: show todos with the specified tag.
- `todo show p/PRIORITY`: show todos with the specified priority.

The header of the todo list panel will also be updated to the condition stated when the todo show command is executed.

### CODE REVIEW

File: src/main/java/soconnect/model/Model.java [hunk 1]
```diff
@@ -231,6 +232,18 @@ public interface Model {
      */
     void updateFilteredTodoList(Predicate<Todo> predicate);
 
+    /**
+     * Returns the header to be displayed on the {@code TodoListPanel}.
+     * @return
```
Comment (Eugene-Ong-W-X) [inline src/main/java/soconnect/model/Model.java]: ?

File: src/main/java/soconnect/model/ModelManager.java [hunk 2]
```diff
@@ -26,6 +27,8 @@ public class ModelManager implements Model {
     private final UserPrefs userPrefs;
     private final FilteredList<Person> filteredPersons;
     private final FilteredList<Todo> filteredTodos;
+    private final ObservableList<Tag> tags;
```
Comment (Eugene-Ong-W-X) [inline src/main/java/soconnect/model/ModelManager.java]: Are you using this? I'm pretty sure @ChongCheeKaiClarence already removed it

File: src/main/java/soconnect/model/todo/Date.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+    private static final DateTimeFormatter DATE_DISPLAY_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy");
+
+    public final LocalDate date;
+
+    /**
+     * Constructs a {@code Date}.
+     *
+     * @param date A valid date that is before today.
```
Comment (Eugene-Ong-W-X) [inline src/main/java/soconnect/model/todo/Date.java]: `before`?

File: src/main/java/soconnect/model/todo/Date.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        }
+
+        return true;
+    }
+
+    /**
+     * Compares this {@code Date} to another {@code Date}.
+     *
+     * @param other The other {@code Date}.
+     * @return Negative if this {@code Date} is earlier and positive if this {@code Date} is latter than the
```
Comment (Eugene-Ong-W-X) [inline src/main/java/soconnect/model/todo/Date.java]: `Negative integer`, `Positive integer`

**Review by Eugene-Ong-W-X — CHANGES_REQUESTED** [review]
Good job, some fixes to take note of

File: src/main/java/soconnect/model/ModelManager.java (hunk 2 above)
Comment (Ugholaf) [reply src/main/java/soconnect/model/ModelManager.java]: Oh I don't need it, I think I didn't remove this properly when resolving the merge conflict. Thank!

**Review by Eugene-Ong-W-X — APPROVED** [review]

File: src/main/resources/view/SoConnectTheme.css [hunk 5]
```diff
@@ -255,6 +255,11 @@
     -fx-vgap: 10;
 }
 
+#todoTags {
+    -fx-hgap: 10;
+    -fx-vgap: 10;
+}
+
```
Comment (florentianayuwono) [inline src/main/resources/view/SoConnectTheme.css]: Sorry, but I think it's better if this group is moved to line 288, after the `.cell_small_label_todo` haha. That way the todo styling is grouped together.

**Review by florentianayuwono — COMMENTED** [review]
Looks great!! Just a small suggestion below, thank you!

File: src/main/resources/view/SoConnectTheme.css (hunk 5 above)
Comment (Ugholaf) [reply src/main/resources/view/SoConnectTheme.css]: Okay done!

**Pull request merged**